src/main/java/by/tms/insta/dao/HibernateUserStorage.java

```diff
@@ -39,4 +40,28 @@ public List<User> findAll() {
         session.close();
         return resultList;
     }
+
+    @Override
+    public boolean userIsExisted(User user) {
+        Session session = sessionFactory.openSession();
```

Comment:
Maybe better to use getCurrentSession(). Please check article https://java2blog.com/difference-opensession-getcurrentsession-hibernate/